app/models/user.rb

```diff
@@ -1,4 +1,16 @@
 class User < ApplicationRecord
   devise :database_authenticatable, :recoverable,
-    :trackable, :validatable, :timeoutable
+    :trackable, :validatable, :timeoutable,
+    :omniauthable, omniauth_providers: [:mojsso]
+
+  def self.from_omniauth(auth)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+      user.provider = auth.provider
+      user.uid = auth.uid
+      user.email = auth.info.email
```

Comment:
Depends on your use case, in PVB `moj-sso` is the only way to sign in so we don't have this scenario. I don't know what is the use case to store the `(provider, uid)` on user...

If you can, making sso the only way to sign in to your app would be the simplest thing to do. For your users (they would only have one account in sso to login) and in the code (you don't even need devise as a dependency).